src/lib/link.ts

```diff
@@ -520,6 +521,39 @@ export class Link {
     return height;
   }
 
+  public async timeoutPackets(
+    source: Side,
+    packets: readonly PacketWithMetadata[]
+  ): Promise<readonly MsgResult[]> {
+    this.logger.info(`Timeout packets for source ${source}`);
+    console.log('N PACKETS:', packets.length);
+    const { src, dest } = this.getEnds(source);
+
+    // check if we need to update client at all
+    const neededHeight = Math.max(...packets.map((x) => x.height)) + 1;
+    // const headerHeight = await this.updateClientToHeight(source, neededHeight);
+    await this.updateClientToHeight(source, neededHeight);
+
+    const [packet] = packets;
+    const nextSequenceRecvResponse = await dest.client.getNextSequenceRecvProof(
```

Comment:
You need the same proof query as for submitting an ack. However, when you get it the value should be empty and the proof existing (it is a non-existence proof) - you can console.log the query to ensure this works